build-logic/buildquality/src/main/kotlin/gradlebuild/testcleanup/TestFilesCleanupService.kt

```diff
@@ -168,11 +176,19 @@ abstract class TestFilesCleanupService @Inject constructor(
 
         // Second run: verify and throw exceptions
         if (!cleanupRunnerStep) {
+            val exceptions = mutableListOf<Exception>()
             parameters.projectExtensions.get()
                 .filter { projectPathToLeftoverFiles.containsKey(it.key) }
                 .forEach { (projectPath: String, projectExtension: TestFileCleanUpExtension) ->
-                    projectExtension.verifyTestFilesCleanup(getFailedTaskPaths(projectPath), projectPathToLeftoverFiles.getValue(projectPath))
+                    try {
+                        projectExtension.verifyTestFilesCleanup(getFailedTaskPaths(projectPath), projectPathToLeftoverFiles.getValue(projectPath))
+                    } catch (e: Exception) {
+                        exceptions.add(e)
+                    }
                 }
+            if (exceptions.isNotEmpty()) {
+                throw DefaultMultiCauseException("", exceptions)
```

Comment:
Would it make sense to re-throw the original exception if there is only one exception?

I think it would also be good to add some message for the multi cause exception here. Are the other exception messages printed when we add an empty message here?

Reply:
Done.